### PR TITLE
Use Z1 and Z1X responses to determine if forms should be uploaded

### PR DESCRIPTION
**Pull request by naomidb**
**What does this change do?** 

These changes rearrange the way the builders run. Every form is pulled out into an individual function. For optional forms, the Z1/Z1X forms are checked to see if the form was actually submitted.

**Why was this change made?**

Previously, if a form, say A2, was not submitted and was marked absent, NACCulator would still create a (blank) A2 form, which would be erroneously uploaded to NACC and have to be manually deleted. These changes will prevent some of that extra, unnecessary work.

## Verification

Check that visits without a certain form in REDCap also do not have said form in the fixed width text file.
Check that outputted text files pass NACC's 

## Affirmations

All of these should have a check by them. Any exception requires an explanation.

* [X] I matched the style of the existing code.
* [X] I added and updated any relevant documentation (inline, `README`, `CHANGELOG`, and such).
* [ ] I used Python's type hinting.
* [X] I ran the automated tests and ensured they **ALL** passed.
* [X] I ran the linter and ensured my changes have not introduced **ANY** warnings or errors.
* [X] I have made an effort to minimize code changes and have not included any cruft (preference files, *.pyc files, old comments, print-debugging, unused variables).
* [X] I have made an effort maintain a clear commit history (haven't merged other branches or rebased improperly).
* [X] I have written the code myself or have given credit where credit is due.

### CODE REVIEW

File: nacc/uds3/fvp/builder.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def build_uds3_fvp_form(record):
     """ Converts REDCap CSV data into a packet (list of FVP Form objects) """
     packet = fvp_packet.Packet()
 
-    # Set up us the forms
```
Comment (ufl-taeber) [inline nacc/uds3/fvp/builder.py]: ;-) https://en.wikipedia.org/wiki/All_your_base_are_belong_to_us

File: nacc/uds3/ivp/builder.py [hunk 2]
```diff
@@ -155,323 +155,338 @@ def add_z1_or_z1x(record, packet):
         packet.insert(0, z1)
 
 
-def add_a1(record,packet):
```
Comment (ufl-taeber) [inline nacc/uds3/ivp/builder.py]: Thank you for adhering to the "no flake8 warnings" rule; this is, however, another instance of where I think it's safe to ignore the warnings as the code is more readable with the aligned `=`'s.

File: nacc/uds3/fvp/builder.py [hunk 3]
```diff
@@ -12,8 +12,140 @@
...
+    add_d1(record, packet)
+    add_d2(record, packet)
+    clsform.add_cls(record, packet, fvp_forms)
+    update_header(record, packet)
+
+    return packet
+
+
+def add_z1_or_z1x(record, packet):
+    # Forms A1, A5, B4, B8, B9, C2, D1, and D2 are all REQUIRED.
```
Comment (s-emerson) [inline nacc/uds3/fvp/builder.py]: it doesn't look like form A5 is required for fvps. it's not in this builder file, nor is it in nacc's fvp ded. this comment also doesn't specify that C1S is required for ptids with Z1 forms, and C2 is required for ptids with Z1X forms.
i was just confused about forms that seemed to be missing, during testing, is all, so the comment (and the fact that it's the exact same comment as in the ivp builder, when there is the difference in A5 forms) felt misleading.

**Comment by ufl-taeber** [comment]
Thanks for testing this out @s-emerson!

**Pull request merged**